cgatcore/database.py

```diff
@@ -45,6 +45,9 @@ def executewait(dbhandle, statement, regex_error="locked",
     while 1:
         try:
             cc = dbhandle.execute(statement)
+        except AttributeError:
+            with dbhandle.begin() as conn:
```

Comment:
the handling of 2.0 is different and does not include a holding loop for locked databases.
to make this completely equivalent, this would probably necessitate another exception handler, although this would look ugly, so any other suggestions would be appreciated.

Reply:
it looks good to me for the time being though?